### ISSUE TITLE
20 Versioning the Application

### ISSUE THREAD

**Issue by Vino-S** [FreeUKGen/MyopicVicar#1268]
Priority 20 (0	6	7	7)

depends on #1251

**Comment by richpomfret** [comment]
@Vino-S  I'd like us to use the year and the sprint (or week) number for the release - which then ties the work completed during the sprint to that specific version. Therefore for the version at the end of this sprint it would be 1747 and the next one after that would be 1749.  It would be nice to get the phase number in there too, however I'm not quite sure how we'd do that unless we use something like 1.4.1749. Bit of a mouthful.  What do you think?

**Comment by richpomfret** [comment]
@Captainkirkdawson For your consideration too (if you get a chance to comment).

**Comment by SteveBiggs** [comment]
The current version on the FR site is FR2.1.5.33 (dated 11 Aug 2017).
Will this become for example, FR2.1.5.1749?

**Comment by richpomfret** [comment]
We can certainly keep the increment/iteration number too. Ideally, I would like to see a consistent release update deployed with every sprint (or can be every 4 weeks if more practical). This will help when it comes to providing release notes (for us and researchers).

**Comment by richpomfret** [comment]
@Captainkirkdawson I would welcome your thoughts here.

**Comment by Captainkirkdawson** [comment]
@richpomfret  as always the devil is in the detail. 
We used versioning in FreeREG for over 2 years. It was a mixed automatic/manual system. 
We used the automatic aspect to document which server was running which version and also to know which version was used to create a specific search record. The latter enabled us to selectively update search records based on how they were created. It also enabled us to know which server was running which version and if we had missed an update (with 5 server that was always and still is possible)
We used the manual aspect to inform the Members on what changes had been made to the application.
The automatic aspect stopped working in July 2017 due to the sharding changes (See active story software versions #1251). The manual aspect stopped when I retired in August. I have no knowledge of the FreeCEN process.

> In August 2017 @Vino-S  suggested using github features
> 1) Create tag -- While we are in the latest stable master branch, create a new tag as 'git tag v1.0.0'  This will create the new tag. I had used Semantic versioning here which is Major.Minor.BugFixes(1.0.0).
> 2) push tag -- Then push the local tag to the repository by, 'git push origin --tags master'
> 3) This will create the tag to the master branch as v1.0.0
> 4) Update this stable v1.0.0 master branch to the production server.
> 5) We can add in our application footer of which version is the application currently running.
> 6) Once we all do the first 5 points, then we work like the usual to create new features, bug fixes and so from taking a new branch from the master.  
> 7) All the newly created branches will be tested and merged into master. In the last day of the Sprint, we use the diff tag of the master branch to find all the changes, bug fixes of the code and create a new tag 'git tag v1.0.1' it will create a new tag. 
> 8) Push the above new tag to the repo and update this branch to the test server. Once the regression and manual testing has been done in the test server then we can update this new version release to the production.                        
> So we can keep the production as stable and add new feature and bug fixing in the local and test environment.

While accepting the suggestion in principle there were issues with the suggestion and responded

> Since the Myopicvicar master branch supports both REG and CEN how can we do it as you suggest?
> FreeREG deploys the master and FreeCEN deploys the freecen parsing branch. 
> From what I understand freecen parsing does not always stay in step with master on a 2 week cycle it has often gone some time before merging master into the freecen parsing branch
> There are also changes and mods made to freecen parsing which never go into master as we do not back merge freecen parsing into master. 
> I am now confused and worried

So are we proposing that freeren parsing be merged into master (quite an undertaking as FreeCEN evolved into using its own version of assets that FreeREG knows nothing about ) and then both apps are deployed from master on a 2 week cycle? This would require that all FreeREG changes are tested in FreeCEN and all FreeCEN changes are tested on FreeREG.  There are not the resources to do that IMO

**Comment by Captainkirkdawson** [comment]
My comments of 3 months ago remain valid. 
This story depends on feature #1251 which is 1 year old!
